File: src/main/java/com/github/tuk-cps/ExprParser/ExprTreeSqrt.java
package ExprParser;

import static jAADD.AADD.REAL;

/**
 * Predefined functions: sqrt
 */
class ExprTreeSqrt extends ExprTreeFunction {
    ExprTreeSqrt()  { super(REAL, "sqrt", new ExprTree("x", REAL)); }
    void evalThis() throws ExprError {
        result = getAADDParam(0).sqrt();
    }
}

File: src/main/kotlin/com/github/tuk-cps/exprParser/ExprTreeSqrt.kt
package exprParser

import jAADD.AADD

/**
 * Predefined functions: sqrt
 */
internal class ExprTreeSqrt : ExprTreeFunction(AADD.Reals, "sqrt", ExprTree(AADD.Reals, "x")) {
    @Throws(ExprError::class)
    override fun evalUp() {
        value = getAADDParam(0).sqrt()
    }
}